src/components/Movie.js

```diff
@@ -5,21 +5,25 @@ import ItemsCarousel from "react-items-carousel";
 import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
 import ArrowBackIosNewIcon from '@mui/icons-material/ArrowBackIosNew';
 import { useDispatch, useSelector } from 'react-redux';
-import {fetchOneMovie} from "../features/imdb.service"
-import { cleanPreviousMovieDetails, getMovieDetails } from '../features/movies/movieSlice';
+import { cleanPreviousMovieDetails, movieSelectors } from '../features/movies/movieSlice';
+import { fetchOneMovie } from '../features/imdb.service';
 
 function Movie() {
     const [activeActor, setActiveActor] = useState(0);
     const [activeSimilar, setActiveSimilar] = useState(0);
     const {id} = useParams();
     const dispatch = useDispatch();
-    const movie = useSelector(getMovieDetails)
-    const [screenSize, setScreenSize] = useState(1090);
+    const [screenSize, setScreenSize] = useState(null);
+    let movie = useSelector((state) => movieSelectors.selectById(state, id));
 
     useEffect(() => {
-        dispatch(fetchOneMovie(id));
+        if(movie.plot){
+            console.log("Have complete info :)");
+        }
+        else{
+            dispatch(fetchOneMovie(id));

```

Comment:
I understand `useEffect` cannot have asynchronous functions, but we can create a function and `await` the `dispatch`. The benefit you get from that is you will be able to show loader. Loaders are component specific so its state shouldn't be in reducer. 

eg.

```JSX
const [isLoading, setLoading] = useState(false);

const fetchMovie = async (id) => {
   setLoading(true);
   const response = await dispatch(fetchOneMovie(id));
   setLoading(false);
}

useEffect(() => fetchMovie(id), [id]);
```